contracts/vault/packages/VaultPackage.sol

```diff
@@ -100,41 +106,33 @@ contract VaultPackage is VaultStorage, IVault, IVaultInit, IVaultEvents {
     }
 
     /// @notice Set the new deposit limit.
-    /// @dev Can not be changed if a depositLimitModule
-    /// is set or if shutdown.
     /// @param _depositLimit The new deposit limit.
-    function setDepositLimit(uint256 _depositLimit) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+    // solhint-disable-next-line code-complexity
+    function setDepositLimitAndModule(uint256 _depositLimit, address _limitModule) external override onlyRole(DEFAULT_ADMIN_ROLE) {
         if (shutdown == true) {
             revert InactiveVault();
         }
-        if (depositLimitModule != address(0)) {
+        if (_depositLimit == 0) {
```

Comment:
This function became a mess. Deposit limit must be > 0 but if limit module isn't 0x than _maxDeposit function will never use it. Previous design actually worked. Also now I see that usage of one module for deposit and withdraw actually doesn't fit the workflow. I think we shall bring back separate deposit and withdraw modules. So other things also need to be brought back or redesigned correctly. If we set deposit limit - we aren't using deposit module. And opposite if we set deposit module - we suppose limit on contract is 0.

Reply:
Fixed on last commit when I bring both depositLimitModule and withdrawLimitModule back